.github/workflows/test.yaml

```diff
@@ -0,0 +1,35 @@
+name: test
+
+on: [pull_request]
```

Comment:
We should think about including `push` in this list.  My understanding is that using the `pull_request` event will try to merge this PR into the base branch during the checkout ([example](https://github.com/NREL/rdtools/runs/1333729136?check_suite_focus=true#step:2:204)), which is usually what we want, but I assume it'll fail/not run when there's a merge conflict.  Adding `push` would mean that a second set of tests run just on the PR branch without trying to merge.  Travis does this ("PR" vs "push").

I think `push` would mean we also get an independent CI run for each commit pushed to master, which I think would be nice.

Reply:
Interesting. We could modify the behavior like this as well: https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit but I think doing both is ultimately what we want like you say.